modules/javafx.graphics/src/main/java/com/sun/prism/impl/PrismSettings.java

```diff
@@ -202,13 +203,17 @@ dirtyOptsEnabled && getBoolean(systemProperties,
 
         String order = systemProperties.getProperty("prism.order");
         String[] tryOrderArr;
+        String[] macDefaultOrderArr = new String[] { "es2", "mtl", "sw" };
+        // macDefaultPipeline flag will be used to determine whether to
+        // use es2/mtl pipeline in glass, when sw pipeline is used in prism
+        macDefaultPipeline = macDefaultOrderArr[0];
```

Comment:
Since this is a platform independent file, should we name this property in a generic way (even if it is only used for MacOS)?
Something like `defaultPipeline?`

Reply:
Since `macDefaultOrderArr` is already mac-specific, I'm not sure how much this helps without also refactoring the rest of the logic.

One possibility would be to move the if-then-else block up (before the check for the `"prism.order"` system property) and then assign the default pipeline after the tryOrder is assigned. I'll leave it up to Jay as whether he wants to do this.